test/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDaoTest.java

```diff
@@ -515,18 +549,23 @@ public void canGetAllSurveys() throws Exception {
         Thread.sleep(GSI_WAIT_DURATION);
         // Get all surveys
         // Make sure this returns all surveys that we created
-        List<Survey> surveys = surveyDao.getAllSurveysMostRecentVersion(TEST_STUDY);
+        List<Survey> surveys = surveyDao.getAllSurveysMostRecentVersion(TEST_STUDY, false);
         assertContainsAllKeys(mostRecentVersionSurveys, surveys);
 
         // Get all surveys of a version
-        surveys = surveyDao.getSurveyAllVersions(TEST_STUDY, survey.getGuid());
+        surveys = surveyDao.getSurveyAllVersions(TEST_STUDY, survey.getGuid(), false);
         assertEquals("All survey versions are returned", 2, surveys.size());
+        int totalCount = surveys.size();
 
         Survey version1 = surveys.get(0);
         Survey version2 = surveys.get(1);
         assertEquals("Surveys have same GUID", version1.getGuid(), version2.getGuid());
         assertEquals("Surveys have same Study key", version1.getStudyIdentifier(), version2.getStudyIdentifier());
         assertNotEquals("Surveys have different createdOn attribute", version1.getCreatedOn(), version2.getCreatedOn());
+        
+        // lets delete an older version and verify that works
+        surveyDao.deleteSurveyPermanently(version1);
+        assertEquals(totalCount-1, surveyDao.getSurveyAllVersions(TEST_STUDY, survey.getGuid(), true).size());
```

Comment:
Similarly, would it be better to iterate the list and verify the deleted version is no longer in the list?

Reply:
Switched to look for the keys of the deleted version specifically